packages/utils/src/promiseQueue.ts

```diff
@@ -44,12 +48,15 @@ export class PromiseQueue<T> {
 
   async run() {
     while (this.total !== this.completed) {
-      if (this.isAborted) break;
+      if (this.isAborted) return;
 
       if (this.runNext()) {
         const promiseFunc = this.todo.shift();
 
-        if (!promiseFunc) break;
+        if (!promiseFunc) {
+          await Promise.allSettled(this.running);
+          return;
+        }
```

Comment:
Due to `return`,`onComplete` function call will not be triggred

Reply:
`onComplete` is not needed if `abort` is called